spec/linkedList.js

```diff
@@ -1,200 +1,72 @@
-        {
-          data:'elementals',
-          next: {
-            data: 'other-elementals',
-            next: null
-          }
-        }
-      )
-    })
+    expect(linkedList).to.be.a('function')
   })
 
   context('insert()', () => {
-    it('Inserts a node (with the provided data) to the tail of the list', () => {
-      const list = new LinkedList()
-
-      expect( list.insert('element') ).to.eql( { data:'element', next: null } )
-    })
-  })
-
-  context('insertFirst()', () => {
-    it('Inserts a node (with the provided data) to the head of the list', () => {
-      const list = new LinkedList()
-
-      list.insert('element')
-      list.insert('other-element')
-      list.insert('another-element')
-      list.insert('elementals')
-      list.insert('other-elementals')
-      list.insertFirst( 'ele' )
-
-      expect( list.getHeadNode().data ).to.eql( 'ele' )
-    })
-  })
-
-  context('insertBefore()', () => {
-    it('Inserts a node (with data "apples") before the first node containing "bananas"', () => {
-      const list = new LinkedList()
-
-      list.insert('element')
-      list.insert('other-element')
-      list.insert('other-elementals')
-      list.insert('elementals')
-      list.insert('bananas')
-
-      // expect( list.size() ).to.eql( 5 )
-      expect( list.insertBefore('bananas', 'apples' ) ).to.eql(
-        { data:'apples', next: { data:'bananas', next: null } }
-      )
-      // expect( list.size() ).to.eql( 6 )
-    })
-  })
-  context('insertAfter()', () => {
-    it('Inserts a node (with data "bananas") after the first node containing "apples"', () => {
-      const list = new LinkedList()
-
-      list.insert('element')
-      list.insert('other-element')
-      list.insert('other-elementals')
-      list.insert('apples')
-      list.insert('elementals')
-
-      // expect( list.size() ).to.eql( 5 )
-      expect( list.insertAfter('apples', 'bananas' ) ).to.eql(
-        { data:'bananas', next: { data:'elementals', next: null } }
-      )
-      // expect( list.size() ).to.eql( 6 )
+    it('creates a new node and places in the linked list as the new tail node', () => {
+      const myLinkedList = new linkedList()
+      myLinkedList.insert( 'water' )
+      myLinkedList.insert( 'balloons' )
+
+      console.log(myLinkedList._length);
+      expect(myLinkedList.head).to.eql({ data: 'water', next: { data: 'balloons', next: null}})
+      expect(myLinkedList._length).to.eql(2)
     })
   })
-
-  context('remove()', () => {
-    it('Removes the tail node from the list', () => {
-      const list = new LinkedList()
-
-      list.insert('element')
-      list.insert('other-element')
-      list.insert('other-elementals')
-      list.insert('apples')
-      list.insert('elementals')
-
-      expect(() => list.remove() ).to.alter(() => list.size(), { from: 5, to: 4 } )
-      expect( list.find( 'elementals' ) ).to.eql( -1 )
-
+  context('getHeadNode()' , () => {
```

Comment:
can you create a test for when there is no HeadNode (the list is empty) to make sure it returns null?